lib/useMap.tsx

```diff
@@ -0,0 +1,45 @@
+import OLMap from "ol/Map";
+import { createContext, useContext, useMemo, useState } from "react";
+
+type ContextType = {
+  map: OLMap | null;
+  // eslint-disable-next-line unused-imports/no-unused-vars
+  setMap: (map: OLMap) => void;
+  removeMap: () => void;
+};
+
+const OLMapContextDefault: ContextType = {
+  map: null,
+  setMap: () => {},
+  removeMap: () => {},
+};
+
+const OLMapContext = createContext<ContextType>(OLMapContextDefault);
+
+type Props = {
+  children: React.ReactNode;
+};
+
+const OLMapProvider = ({ children }: Props) => {
+  const [map, setMap] = useState<OLMap | null>(null);
+
+  const value = useMemo(() => ({ map, setMap, removeMap: () => {} }), [map]);
+
+  return (
+    <OLMapContext.Provider
+      value={value}
+    >
+      {children}
+    </OLMapContext.Provider>
+  );
+};
+
+const useMap = () => {
```

Comment:
`useMap` might be confusing for new joiners since React already have a hook named `useMap`.

For the ease of readability name would be `useOlMap` or something else